### DISCUSSION TITLE
My function is not completely compiled. What is missing?

### DISCUSSION THREAD

**Discussion by FranciscoOssian** [emscripten-core/emscripten#17262] [Q&A] [answered]
```
//main/c
#include <stdio.h>
#include <emscripten.h>
EMSCRIPTEN_KEEPALIVE
void run() {
	int stop = 100;
  	for(;stop >= 0; --stop){
  		printf(" value ->  %i\n", stop);
  	}
	printf("hi");
}
```
```
//main.js
var factory = require('./api_example.js');
factory().then((instance) => {
  instance._run(); // direct calling works
});
```

`emcc main.c -o api_example.js -sMODULARIZE -sEXPORTED_RUNTIME_METHODS=ccall`

It prints "value -> X", but never "hi". deleting emscript dependencies in main.c and use gcc "hi" is printed

**Comment by sbc100** [comment, marked as answer]
The final printf is lacking a newline.. which means its most likely buffered in the stdout line buffer.    Try adding a newline or calling `fflush()`.

**Reply by FranciscoOssian** [reply]
Yep, sorry. I wasn't remembering this.
